packages/eslint-plugin/src/rules/thenable-in-promise-aggregators.ts

```diff
@@ -0,0 +1,258 @@
+      for (const declaration of declarations) {
+        const sourceFile = declaration.getSourceFile();
+        if (program.isSourceFileDefaultLibrary(sourceFile)) {
+          return true;
+        }
+      }
+
+      return false;
+    }
+
+    // Is like Promise, or a class/interface extending from Promise
+    function isPromiseClassLike(program: ts.Program, type: ts.Type): boolean {
+      if (type.isIntersection()) {
+        return type.types.some(t => isPromiseLike(program, t));
+      }
+      if (type.isUnion()) {
+        return type.types.every(t => isPromiseLike(program, t));
+      }
+
+      const symbol = type.getSymbol();
+      if (!symbol) {
+        return false;
+      }
+
+      if (
+        symbol.getName() === 'Promise' &&
+        isSymbolFromDefaultLibrary(program, symbol)
+      ) {
+        return true;
+      }
+
+      if (symbol.flags & (ts.SymbolFlags.Class | ts.SymbolFlags.Interface)) {
+        const checker = program.getTypeChecker();
+
+        for (const baseType of checker.getBaseTypes(type as ts.InterfaceType)) {
+          if (isPromiseLike(program, baseType)) {
+            return true;
+          }
+        }
+      }
+
+      return false;
+    }
+
+    // Is like PromiseConstructor, or a class/interface extending from Promise
+    function isPromiseLike(program: ts.Program, type: ts.Type): boolean {
+      if (type.isIntersection()) {
+        return type.types.some(t => isPromiseLike(program, t));
+      }
+      if (type.isUnion()) {
+        return type.types.every(t => isPromiseLike(program, t));
+      }
+
+      const symbol = type.getSymbol();
+      if (!symbol) {
+        return false;
+      }
+
+      if (
+        symbol.getName() === 'PromiseConstructor' &&
+        isSymbolFromDefaultLibrary(program, symbol)
+      ) {
+        return true;
+      }
+
+      if (symbol.flags & (ts.SymbolFlags.Class | ts.SymbolFlags.Interface)) {
+        const checker = program.getTypeChecker();
+
+        for (const baseType of checker.getBaseTypes(type as ts.InterfaceType)) {
+          if (isPromiseClassLike(program, baseType)) {
+            return true;
+          }
+        }
+      }
+
+      return false;
+    }
+
+    return {
+      CallExpression(node: TSESTree.CallExpression): void {
+        const callee = skipChainExpression(node.callee);
+        if (callee.type !== AST_NODE_TYPES.MemberExpression) {
+          return;
+        }
+
+        if (callee.computed) {
+          if (
+            callee.property.type !== AST_NODE_TYPES.Literal ||
+            typeof callee.property.value !== 'string' ||
+            !aggregateFunctionNames.includes(callee.property.value)
+          ) {
+            return;
+          }
+        } else if (!aggregateFunctionNames.includes(callee.property.name)) {
+          return;
+        }
+
+        const callerType = services.getTypeAtLocation(callee.object);
+        if (!isPromiseLike(services.program, callerType)) {
```

Comment:
```suggestion
        const calleeType = services.getTypeAtLocation(callee.object);
        if (!isPromiseLike(services.program, calleeType)) {
```

[nit] let's name it `calleeType` instead of `callerType` since it is named `callee` elsewhere